dockerfile_parse/parser.py

```diff
@@ -767,7 +767,8 @@ def add_lines(self, *lines, **kwargs):
         for stage in range(len(froms)-2, -1, -1):  # e.g. 0 for single or 2, 1, 0 for 3 stages
             start, finish = froms[stage], froms[stage+1]
             linenum = start['endline'] + 1 if at_start else finish['startline']
-            if skip_scratch and froms[stage]['value'] == 'scratch':
+            if skip_scratch and (froms[stage]['value'] == 'scratch' or
+                                 froms[stage]['value'].lower().startswith('scratch as')):
```

Comment:
:( this makes me sad, so this parser doesn't recognize FROM value and stage name?

how about `scratch  as`, `scratch        as`, etc ...

Reply:
https://github.com/containerbuildsystem/dockerfile-parse/blob/master/dockerfile_parse/parser.py#L868 can we resue this to avoid duplicated logic?

Reply:
I reused `image_from` PTAL.